newIDE/app/src/Leaderboard/LeaderboardProvider.js

```diff
@@ -122,13 +125,30 @@ const reducer = (state: ReducerState, action: ReducerAction): ReducerState => {
         displayOnlyBestEntry: action.payload,
       };
     case 'UPDATE_OR_CREATE_LEADERBOARD':
+      let leaderboardsByIdsWithUpdatedPrimaryFlags;
+      if (!state.leaderboardsByIds) {
+        leaderboardsByIdsWithUpdatedPrimaryFlags = {
+          [action.payload.id]: action.payload,
+        };
+      } else {
+        leaderboardsByIdsWithUpdatedPrimaryFlags = {};
+        Object.entries(state.leaderboardsByIds).forEach(
+          ([leaderboardId, leaderboard]) => {
+            leaderboardsByIdsWithUpdatedPrimaryFlags[leaderboardId] = {
+              ...leaderboard,
+              // $FlowFixMe: known error where Flow returns mixed for object value https://github.com/facebook/flow/issues/2221
+              primary: action.payload.primary ? undefined : leaderboard.primary,
+            };
+          }
+        );
+        leaderboardsByIdsWithUpdatedPrimaryFlags[action.payload.id] =
```

Comment:
you're doing the same thing twice (in `if` and `else`) this could be outside of if/else? (and reverse your if logic)

Reply:
Indeed, that's possible! Thanks
I was paranoid that Flow wouldn't accept it, I feel like TS wouldn't have accepted it (you set keys in an untyped object)